src/components/image/index.tsx

```diff
@@ -1,15 +1,39 @@
 import React, { ImgHTMLAttributes } from 'react';
-
-import { getSource } from '../../util/image/util';
+import { AdvancedImage } from '@cloudinary/react';
+import { CloudinaryImage } from '@cloudinary/url-gen';
+import { getCloudinaryImage, getHashFromIpfsUrl } from '../../util/image/util';
 
 export interface Properties extends ImgHTMLAttributes<HTMLImageElement> {
   className: string;
+  useCloudinary?: boolean;
+  cloudinaryTransformable?: (
+    cloudinaryImage: CloudinaryImage
+  ) => CloudinaryImage;
 }
 
 export default class Image extends React.Component<Properties> {
+  renderCloudinaryImage = (): React.ReactElement => {
+    const { className, src, cloudinaryTransformable, ...rest } = this.props;
+    const hashImage = getHashFromIpfsUrl(src);
+
+    let cloudinaryImage = getCloudinaryImage(hashImage);
```

Comment:
if we *really* want to be service agnostic (i don't think it matters toooo much at this point, but would also make it a bit easier to test), we could pull these things into a separate library or component. it can probably wait, though, until we add another transformation option.